### PR TITLE
Export Transitioner for react-navigation

### PR DESCRIPTION
**Pull request by fangpenlin**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by fangpenlin** [comment]
Yo hey, first time contributor here, I am still checking to see if anything is wrong. Let me know if there is anything needs to be changed.

**Comment by fangpenlin** [comment]
Context for this PR:

[Transitioner](https://reactnavigation.org/docs/views/transitioner) is one important view for implementing custom animation, but it's not exported in the declare file, this PR simply adds the missing Transitioner.

**Comment by huhuanming** [comment]
cool, all is well.
And you should complete unchecked things

image

**Comment by fangpenlin** [comment]
done

**Comment by fangpenlin** [comment]
@huhuanming done

File: types/react-navigation/index.d.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 // Type definitions for react-navigation 1.0
 // Project: https://github.com/react-community/react-navigation
-// Definitions by: Huhuanming <https://github.com/huhuanming>, mhcgrq <https://github.com/mhcgrq>
+// Definitions by: Huhuanming <https://github.com/huhuanming>
+//                 mhcgrq <https://github.com/mhcgrq>
+//                 fangpenlin <https://github.com/fangpenlin>
```
Comment (minestarks) [inline types/react-navigation/index.d.ts]: Could you revert this to the single-line comma-separated format? I think @dt-bot has trouble parsing headers that deviate at all from this exact format.

**Comment by minestarks** [comment]
@fangpenlin please see comment about header. Looks good otherwise

**Comment by minestarks** [comment]
@fangpenlin my mistake, header is actually fine. Thanks for the PR!

**Pull request merged**